lib/ActivityStreams.js

```diff
@@ -152,9 +152,11 @@ ActivityStreams.prototype = {
    * Process the passed in links, save them, get from cache and response to content.
    */
   _processAndSendLinks(links, responseType, append, worker, previewsOnly = false) {
+    const event = this._tabTracker.generateEvent({source: responseType});
     let processedLinks = this._previewProvider.processLinks(links);
-    this._previewProvider.asyncSaveLinks(processedLinks);
+    this._previewProvider.asyncSaveLinks(processedLinks, event);
```

Comment:
A bit of a nit, but a lot of the other methods specify `event` First and other args after. Not sure if worth reordering.